content/articles/introducing-threads-in-java-socket-programming/index.md

```diff
@@ -0,0 +1,442 @@
+            // establishing the connection 
+            Socket newsoct = new Socket(adress, 3333);           
+            DataInputStream sdtinput = new DataInputStream(newsoct.getInputStream());
+            DataOutputStream sdtOutput = new DataOutputStream(newsoct.getOutputStream());
+    // The following loop is responsible for exchanging data between the client and client handle.
+            while (true)
+            {
+                System.out.println(sdtinput.readUTF());
+                String tosend = newscanner.nextLine();
+                sdtOutput.writeUTF(tosend);
+                // If the client sends an exit, then the connection should be closed and the while loo should be terminated.
+                if(tosend.equals("Exit"))
+                {
+                    System.out.println("Closing connection : " + newsoct);
+                    newsoct.close();
+                    System.out.println("Closed");
+                    break;
+                }
+                
+                
+                String newresuiltReceived = sdtinput.readUTF();
+                System.out.println(newresuiltReceived);
+            }
+            
+            newscanner.close();
+            sdtinput.close();
+            sdtOutput.close();
+        }catch(Exception e){
+            e.printStackTrace();
+        }
+    }
+}
+
+ ```
+ ### Program's method of operation
+To run the Java Server Socket Program, you must first launch it from the data output stream prompt (console). Then, you will see a message that reads "Server Started..." on your data output stream screen.
+
+Java Client Socket Programs must be opened on both computers or on the same network. Your client program will wait for your input when you start it up. If you input a message and press ENTER, the server will also data input stream play the message. You can send messages to the client from the server after receiving them from the client. When the client transmits the word "bye" from the client-side of the connection, the connection is terminated.
+### How these programs operate together
+- A new thread is started to connect to the server for each request made by a client. To communicate with the client, the newly assigned thread is given access to the available streams.
+
+- It is accepted by the server in the while loop after it has been assigned.
+
+- As long as the first request is still being processed, the server accepts and processes the second one. Even as other requests are being processed, many requests can be addressed simultaneously.
+### Testing the above program 
+It does thread matter where you save the two applications. Afterward, run the Server.java and the Client.java one after the other. In order to execute numerous instances of the client program at the same time, utilize an IDE like multiple instances, or run the client software in separate files. To get the results data input stream played above, you'll need numerous clients.

```

Comment:
reword to "It does thread matter where you save the two applications. Afterward, run the Server.java and the Client.java one after the other. To execute numerous instances of the client program simultaneously, utilize an IDE like multiple instances, or run the client software in separate files. To get the results data input stream played above, you will need numerous clients."